internal/app/machined/pkg/runtime/v1alpha1/v1alpha1_sequencer.go

```diff
@@ -413,10 +418,6 @@ func (*Sequencer) StageUpgrade(r runtime.Runtime, in *machineapi.UpgradeRequest)
 		).Append(
 			"dbus",
 			StopDBus,
-		).AppendWhen(
```

Comment:
Do we call LeaveEtcd somewhere else?
Otherwise it shouldn't be dropped, but the condition should be just `r.Config().Machine().Type() != machine.TypeWorker`

Reply:
yes, we do call LeaveEtcd from other flows (reset), but upgrade no longer nukes the system disk, so no need to leave